src/honeybot/plugins/downloaded/duckduckgo/main.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+"""
+[duckduckgo.py]
+Duckduckgo Plugin
+
+[Author]
+Rakesh Seal
+
+[About]
+Returns a abstract of the search query from a duckduckgo Search. Return empty if no abstract found.
+[Website]
+
+
+[Commands]
+>>> .duck <query>
+returns abstrack from duckduckgo search
+"""
+
+import requests
+
+
+class Plugin:
+    def __init__(self):
+        pass
+    
+    def __gen_url_from_query(self, query):
+        return "https://duckduckgo.com/?q=" + query + "&format=json&pretty=1"
+
+    def run(self, incoming, methods, info, bot_info):
+        # if '!~' in info['prefix']:
+        # print(info)
+        try:
+            msgs = info["args"][1:][0].split()
+            if info['command'] == 'PRIVMSG' and msgs[0] == '.duck':
+                    query = msgs[1]
```

Comment:
This works but there is a limitation, for example:

If one messages 
```python
# .duck operating system
msgs[1] == "operating"
msgs[2] == "system"
```

Perhaps we can do a
```python
query = ''.join(msg for msg in msgs[1:])
```
This should add the messages following the `.duck` command together so `query == "operating system".